resources/showIcons.php

```diff
@@ -58,7 +58,20 @@ function showFiles() {
     
     sort($list);
     foreach ($list as $entry) {
-        echo '<tr><td>'.$entry.' </td><td bgcolor="#C0C0C0"><a href="'.$entry.'"><img src="'.$entry.'"></a></td>'."\n";
+        echo '<tr>'."\n";
+        echo     '<td><a href="'.$entry.'">'.$entry.' </a></td>'."\n";
+       if (endswith($entry, ".gif") 
+            || endswith($entry, ".jpg") 
+            || endswith($entry, ".png")
+            ) {
+                // display as image
+                //   would be good to add a limiting size here
+                echo     '<td bgcolor="#C0C0C0"><a href="'.$entry.'"><img src="'.$entry.'"></a></tr>'."\n";
```

Comment:
Sorry, should have said 'if you leave out height and width'.

Having 'max-height' along with 'max-width' will use those as constraints but will preserve aspect ratio.

OK, the image file is still loaded full-size - this is just in-line CSS styling.